amethyst_window/src/bundle.rs

```diff
@@ -53,19 +50,34 @@ impl WindowBundle {
 }
 
 impl SystemBundle for WindowBundle {
-    fn build(
-        self,
+    fn load(
+        &mut self,
         _world: &mut World,
-        _resources: &mut Resources,
-        builder: &mut DispatcherBuilder<'_>,
+        resources: &mut Resources,
+        builder: &mut DispatcherBuilder,
     ) -> Result<(), Error> {
-        let event_loop = EventsLoop::new();
+        let events_loop = EventsLoop::new();
 
-        builder.add_system(
-            Stage::Begin,
-            build_window_system_from_config(&event_loop, self.config),
-        );
-        builder.add_thread_local_system(build_events_loop_system(event_loop));
+        let window = self
+            .config
+            .clone()
+            .into_window_builder(&events_loop)
+            .build(&events_loop)
+            .expect("Unable to create window");
+
+        let hidpi = window.get_hidpi_factor();
+        let (width, height) = window
+            .get_inner_size()
+            .expect("Window closed during initialization!")
+            .to_physical(hidpi)
+            .into();
+
+        resources.insert(ScreenDimensions::new(width, height, hidpi));
```

Comment:
A window can be dragged onto another monitor with different dimensions. Is this handled elsewhere?

Reply:
Nothing has changed here functionality wise from specs, only resource initialization was shuffled around. `ScreenDimensions` are updated from the `build_window_system`.

Reply:
Then this might be a bug, but not in the scope of this PR